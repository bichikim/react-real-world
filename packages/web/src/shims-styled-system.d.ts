import {ConfigStyle} from 'styled-system/index.d.ts'
import {PureObject, SystemFunction} from './types'

declare module 'styled-system' {
  export * from 'styled-system/index.d.ts'

  export type Config2<P extends string> = Record<P, ConfigStyle | boolean>

  export function system<P extends PureObject>(styleDefinitions: Config2<keyof P>): SystemFunction<P>
}
