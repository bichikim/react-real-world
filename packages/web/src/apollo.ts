import {ApolloClient, createHttpLink, InMemoryCache} from '@apollo/client'
import {FC} from 'react'
import {apiUrl} from 'src/env'

/**
 * fix ApolloProvider type error
 */
declare module '@apollo/client' {
  export interface ApolloProviderProps<TCache> {
    client: ApolloClient<TCache>
  }
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  export const ApolloProvider: FC<ApolloProviderProps<any>>
}

export const createClient = (uri: string) => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: createHttpLink({
      uri,
    }),
    ssrMode: true,
  })
}

/**
 *
 */
export const client = createClient(apiUrl())
