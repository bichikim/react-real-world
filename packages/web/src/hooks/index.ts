export * from './init'
export * from './fast-memo'
export * from './debounce'
export * from './ref-act'
