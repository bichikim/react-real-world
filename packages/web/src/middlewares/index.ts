import csp from './csp'
import securityHeaders from './security-headers'
import cookie from './cookie'

export {csp, securityHeaders, cookie}
