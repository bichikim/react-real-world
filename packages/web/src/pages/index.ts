import {MainPage} from 'src/views/main'

export default MainPage

export const getServerSideProps = () => {
  return {
    props: {},
  }
}
