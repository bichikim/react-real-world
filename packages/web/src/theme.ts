import {Theme} from 'styled-system'

export const defaultTheme: Theme = {
  fontSizes: Object.assign([], {lg: '24px', md: '20px', sm: '16px', xl: '28px'}),
}
