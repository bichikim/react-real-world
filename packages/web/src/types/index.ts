export * from './language'
export * from './system'
