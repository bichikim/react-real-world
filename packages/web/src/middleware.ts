import {NextPageContext} from 'next'
import {IS_SSR} from 'src/utils'

export type MiddlewareContext = Required<Omit<NextPageContext, 'err' | 'asPath'>> &
Pick<NextPageContext, 'err' | 'asPath'>

export interface Middleware {
  match?: RegExp

  (
    context: Required<Omit<NextPageContext, 'err' | 'asPath'>> &
    Pick<NextPageContext, 'err' | 'asPath'>,
  ): Promise<any> | any
}

export function matchAndRun(path: string, context: MiddlewareContext, middleware: Middleware) {
  if (typeof middleware !== 'function') {
    if (process.env.NODE_ENV === 'development') {
      console.error('[middlewares] A middlewares must be a function')
    }
    return null
  }

  const {match} = middleware

  if (match) {
    if (match.test(path)) {
      return middleware(context)
    }
    return null
  }
  return middleware(context)
}

/**
 * run middlewares
 * @param middlewares
 * @param context
 */
export const middleware = async (
  middlewares: Record<string | symbol, Middleware>,
  context: NextPageContext,
): Promise<any> => {

  const {pathname} = context

  const {req, res, ...rest} = context

  if (!IS_SSR || !req || !res) {
    return Promise.resolve({})
  }

  const _context = {req, res, ...rest}

  const keys = Object.keys(middlewares)

  const result = await Promise.all(
    keys.map((key) => {
      const middleware = middlewares[key]

      return matchAndRun(pathname, _context, middleware)
    }, []),
  )

  return result.reduce((result, item, index) => {
    if (!item) {
      return result
    }

    const key = keys[index]
    result[key] = item
    return result
  }, {})
}

export default middleware
