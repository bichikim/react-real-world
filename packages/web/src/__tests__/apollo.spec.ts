import {client} from 'src/apollo'
import {ApolloClient, gql} from '@apollo/client'
import fetchMock from 'jest-fetch-mock'
import {apiUrl} from 'src/env'

describe('apollo', function test() {
  beforeEach(() => {
    fetchMock.resetMocks()
  })

  describe('client', function test() {
    it('should be apollo client', function test() {
      expect(client).toBeInstanceOf(ApolloClient)
    })
    it('should fetch query', async function test() {
      const query = gql`
        query {
          foo {
            foo
          }
        }
      `
      fetchMock.mockResponse(JSON.stringify({
        data: {
          foo: {
            foo: 'foo',
          },
        },
      }))
      await client.query({query})
      const fetchMockCalls = fetchMock.mock.calls
      expect(fetchMockCalls.length).toBe(1)
      const url = fetchMockCalls[0][0]
      const request = fetchMockCalls[0][1]
      expect(url).toBe(apiUrl())
      expect(request.method).toBe('POST')
      expect(request.body).toMatchSnapshot()
    })
  })
})
