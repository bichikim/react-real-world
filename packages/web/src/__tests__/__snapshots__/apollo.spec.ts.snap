// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apollo client should fetch query 1`] = `"{\\"variables\\":{},\\"query\\":\\"{\\\\n  foo {\\\\n    foo\\\\n    __typename\\\\n  }\\\\n}\\\\n\\"}"`;
